client/core/core.go

```diff
@@ -7511,7 +7524,22 @@ func (c *Core) handleConnectEvent(dc *dexConnection, status comms.ConnectionStat
 	topic := TopicDEXDisconnected
 	if status == comms.Connected {
 		topic = TopicDEXConnected
+		dc.lastConnectMtx.Lock()
+		dc.lastConnect = time.Now()
+		dc.lastConnectMtx.Unlock()
 	} else {
+		dc.lastConnectMtx.RLock()
+		lastConnect := dc.lastConnect
+		dc.lastConnectMtx.RUnlock()
+		if atomic.LoadUint32(&dc.anomaliesCount)%wsMaxAnomalyCount == 0 {
+			// Send notification to check connectivity.
+			subject, details := c.formatDetails(TopicDexConnectivity, dc.acct.host)
+			c.notify(newConnEventNote(TopicDexConnectivity, subject, dc.acct.host, dc.status(), details, db.Poke))
+		} else if time.Since(lastConnect) < wsAnomalyDuration {
+			// Increase anomalies count for this connection.
+			atomic.AddUint32(&dc.anomaliesCount, 1)
+		}
```

Comment:
Actually, I wonder if it should be something like
```Go
dc.lastConnectMtx.RLock()
lastConnect := dc.lastConnect
dc.lastConnectMtx.RUnlock()
if time.Since(lastConnect) < wsAnomalyDuration {
	// Increase anomalies count for this connection.
	count := atomic.AddUint32(&dc.anomaliesCount, 1)
	if count%wsMaxAnomalyCount == 0 {
		// Send notification to check connectivity.
		subject, details := c.formatDetails(TopicDexConnectivity, dc.acct.host)
		c.notify(newConnEventNote(TopicDexConnectivity, subject, dc.acct.host, dc.status(), details, db.Poke))
	}
} else {
	atomic.StoreUint32(&dc.anomaliesCount, 0)
}
```

Reply:
> atomic.StoreUint32(&dc.anomaliesCount, 0)

@chappjc, had suggested we should not reset to zero https://github.com/decred/dcrdex/pull/2028#discussion_r1089527870. But I just noticed there is something I was reminded of rn.

Reply:
That comment was to avoid a sequential (non-atomic) load/store that you previously had depending on `wsMaxAnomalyCount`.  Resetting to zero based on time since last anomaly is a different matter.

Reply:
>  Resetting to zero based on time since last anomaly is a different matter.

I agree.  I was referring to the part of your comment that mentioned  `(never reset to 0)` but maybe since that was in a different context I might be wrong.

Reply:
Right, different context.  Before it was just about wrapping the counter when a modulo would do the job.  Here's it's about resetting the counter when it has been a long time since last fail.

Reply:
#2235 is up to address a nit, do y'all want these changes there?

Reply:
> #2235 is up to address a nit, do y'all want these changes there?

Up to @buck54321, but that would be the time and place if it makes sense and tests well.

Reply:
Let's do it

Reply:
Done.